daemon/cluster/services.go

```diff
@@ -111,7 +111,8 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string) (*apity
 			digestImage, err := c.imageWithDigestString(ctx, ctnr.Image, authConfig)
 			if err != nil {
 				logrus.Warnf("unable to pin image %s to digest: %s", ctnr.Image, err.Error())
-				resp.Warnings = append(resp.Warnings, fmt.Sprintf("unable to pin image %s to digest: %s", ctnr.Image, err.Error()))
+				// warning in the client response should be concise
+				resp.Warnings = append(resp.Warnings, fmt.Sprintf("image %s could not be accessed on a registry to record its digest. Each node will access %s independently, possibly leading to different nodes running different versions of the image.", ctnr.Image, ctnr.Image))
```

Comment:
Let's define a function that returns this, so we don't have to duplicate the message in two places.

It would be nice to wrap the lines (i.e. to 80 characters), because otherwise some words will be split between lines, which can be hard to read. It's a little hard to do in this case because `ctnr.Image` is variable-length. Maybe we could borrow the code from https://github.com/mitchellh/go-wordwrap. I know this is getting a little complicated for clarifying a warning though...